doc/tools/coqrst/coqdomain.py

```diff
@@ -785,6 +785,47 @@ def run(self):
         self.options['classes'] = ['admonition', 'note']
         return super().run()
 
+class KnownIssueDirective(BaseAdmonition):
+    """A reST directive for known issues.
+
+    This directive is used to document known issues (confirmed bugs or
+    limitations) that do not look like they will be fixed in time for
+    the next release.
```

Comment:
"may" does not sound specific enough, do you have another suggestion? My goal is to say that (in the future) when we are aware of an issue, we should always do one of the following in time for the next release: fix it or document it. Of course, fixing it is better so if a fix is being worked on, documenting can wait, but if we discover that the fix will not be ready in time, then we should definitely document it.

> In practice, for many of the issues, the question is realistically whether they will _ever_ get fixed.

That's right, and one of the next step will be to propose a plan to close known issues that are documented and not actively worked on (keeping the possibility of reopening them at any time if they raise new discussion or activity).

Reply:
Well, "may not" means the same as yours but more succinct.  "won't" would remove the uncertainty that your phrase had.

> My goal is to say that (in the future) when we are aware of an issue, we should always do one of the following in time for the next release: fix it or document it.

An excellent goal.  I hope people will step forward to fix things.  There's also a considerable backlog to review/document--2500 open issues.  Do you think we should include enhancement requests in the doc?

> one of the next step will be to propose a plan to close known issues that are documented and not actively worked on

How about instead tagging issues that have been documented?  (Though I don't see a way to search for all issues that don't tag X set, which would be quite handy.)

Reply:
Or maybe tag them as documented before closing them.

Reply:
> Do you think we should include enhancement requests in the doc?

The scope is a bit hard to define, but probably only enhancement requests that overcome known limitations.

>How about instead tagging issues that have been documented? (Though I don't see a way to search for all issues that don't tag X set, which would be quite handy.)

My idea is to use one of these new GitHub projects for that: https://github.com/orgs/coq/projects/3/views/4.
I've just published a blog post about the whole proposal: https://www.theozimmermann.net/2021/11/known-issues/

Reply:
I didn't feel like subscribing to Disqus, so I'll comment on your blog post here:

Some good ideas here.  A couple thoughts:

- I think the tag model for issues offered by GitHub isn't quite rich enough.  It should be possible to organize issues hierarchically.  For example, there should be a grouping for all bugs related to tactics.  Users should be able to browse the hierarchy with a tree control.  And perhaps it should be a more general graph rather than just a DAG or tree.  Maybe the ability to associate bugs with specific source files and to readily find such bugs given the file name would help.  Also would be good to have some simple workflow, e.g. all new issues are marked "needs: triage" and have very easy ways to see what bugs have not been triaged and how long they've been pending.

- When an issue is added, maybe it would be interesting to systematically add test cases for them illustrating the issue.  Could be helpful in making a fix, also it would be harder to miss that something is fixed.

Reply:
Actually, after subscribing, I still don't see how to publish my comments.  (Just observing, not asking for help.)